staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -146,6 +146,7 @@ type userAgentRoundTripper struct {
 	rt    http.RoundTripper
 }
 
+// NewUserAgentRoundTripper will add User-Agent property to a request unless it has already been set
```

Comment:
I'm not sure I get the "unless it has already been set" part.

Trying to understand the code here. Looks like it uses the User-Agent from the request, and if it's already got one it uses it, otherwise it's ignored.

I would maybe break the description into two sentences:
```
NewUserAgentRoundTripper will add User-Agent header to requests of rt. agent is ignored if the Header is already set on the request.
```

Reply:
I'm following the conventions which were used for other `NewFooRoundTripper`, which uses the "unless it has already been set".

that said, I agree that it might be better with emphasis on "ignore if x". I'm open to hearing more thoughts from folks (especially from SIG API Machinery) on their stance / consistency. Perhaps there is a blanket documentation on these functions?

Reply:
nit: s/property/header
and please add a period at the end of the comment.

What causes the confusion in the original comment "unless it has already been set"?

Reply:
fixed  `s/property/header/` and period at the end in 889ffb0.

I think it's more likely for me to need a double-take when a sentence is formed as "do x, unless y holds" instead of "favors x, fallbacks to y" -- so in this case I personally would find it more settling to read:
> NewUserAgentRoundTripper honors existing header in `rt http.RoundTripper`, fallbacks to provided `agent string`

I'm still learning this code base, so please let me know if this drifts from the original intention 😄